File: dotnet/cafdemalihapa/Applications/NotepadPlusPlusFileManagement/CloseProcessManagement.cs
using System.Diagnostics;
using cafdemalihapa.Directories.HostingDirectory.FeatureDirectory.AutomationsDirectory.ProcessesMetaDataDirectory;
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Applications.NotepadPlusPlusFileManagement
{
    public class CloseProcessManagement(
        ILogger<CloseProcessManagement> logger,
        ProcessesMetaDataDirectory processesMetaDataDirectory
        )
    {
        public void Run()
        {
            var notepadPlusPlusFileProcessesMetaDataDirectory = Path.Combine(processesMetaDataDirectory.GetPath(), "notepad-plus-plus-file-processes-meta-data.json");

            using StreamReader r = new(notepadPlusPlusFileProcessesMetaDataDirectory);
            var json = r.ReadToEnd();
            var readProcessInformationGroup = Newtonsoft.Json.JsonConvert.DeserializeObject<ProcessInformationGroup>(json);

            logger.LogInformation("items, {items}", readProcessInformationGroup);

            foreach (var pInfo in readProcessInformationGroup?.Group)
            {
                logger.LogInformation("Id: {Id}", pInfo?.Id);
                var p = Process.GetProcessById((int)pInfo?.Id);
                p.Kill();
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/NotepadPlusPlusFileManagement/CloseProcessManagement.fs
namespace cafdemwimu.console.Applications.NotepadPlusPlusFileManagement

open System.Diagnostics
open System.IO
open Microsoft.Extensions.Logging
open cafdemwimu.console.Directories.Hosting.Feature.Automations.ProcessesMetaData

type CloseProcessManagement
    (
        logger: ILogger<CloseProcessManagement>,
        processesMetaDataDirectory: ProcessesMetaDataDirectory
    ) =
    member _.Run() =
        let notepadPlusPlusFileProcessesMetaDataDirectory = Path.Combine(processesMetaDataDirectory.GetPath(), "notepad-plus-plus-file-processes-meta-data.json")

        let json =
            use r = new StreamReader(notepadPlusPlusFileProcessesMetaDataDirectory)
            r.ReadToEnd()
        let readProcessInformationGroup = Newtonsoft.Json.JsonConvert.DeserializeObject<ProcessInformationGroup>(json)

        logger.LogInformation("items, {items}", readProcessInformationGroup)

        if not (isNull readProcessInformationGroup) then
            for pInfo in readProcessInformationGroup.Group do
                logger.LogInformation("Id: {Id}", pInfo.Id)
                let p = Process.GetProcessById(pInfo.Id.GetValueOrDefault(0))
                p.Kill()
